bebraven/platform
# Add UI, controller for Survey Submissions

### Description

This adds the controller and views to submit impact surveys.

### Motivation and Context

This PR makes a couple of big assumptions:

1. We have a way of inserting a survey version into a Canvas course (e.g., the `BaseCourseCustomContentVersion` exists between a course (not a template) and survey version).
2. We have a way of generating the HTML `body` with `<input name="unique_name">` using CKE.

I did this all manually to test this PR, including changing the CKE output in the HTML tab in the Content Editor for the Survey I used.

This is the HTML:

```
<h5 id="6"><a href="#6">Note: If you were absent from this Learning Lab, please do not complete the survey -- it will not count against you.</a></h5><p>Do you have any feedback on today's Learning Lab? Things that could improved, or anything you feel confused about? Feedback about your Leadership Coach?</p><p><input id="unique_id_1" data-bz-retained="retained_43_1" type="text" placeholder=""></p>
```

This PR does **not** update the Canvas assignment after submission, there's a TODO task for it: https://app.asana.com/0/1174274412967132/1198971448730205. We'll add the this once we have the Canvas assignment ID/LTI launch information necessary to determine which Canvas assignment the submission is for. This requires we have the code in for adding Surveys to a Course Template done, which @sadleb has started in #412. 

### Test

- Manually tie a `SurveyVersion` to a `Course` by adding it to `BaseCourseCustomContentVersion` (see PR #403 for how to do this in the Rails console)
- Go to `https://platformweb/base_course_custom_content_versions/{id}/survey_submissions/new` where the `id` is for the `BaseCourseCustomContentVersion` you just added, see the empty survey submission form:
<img width="1792" alt="Screen Shot 2020-10-28 at 7 48 49 PM" src="https://user-images.githubusercontent.com/62911934/97520113-b8e81a00-1957-11eb-8a0f-8bda6e01d041.png">

- Type something in the input field:
<img width="1792" alt="Screen Shot 2020-10-28 at 7 49 02 PM" src="https://user-images.githubusercontent.com/62911934/97520149-c7cecc80-1957-11eb-9327-bdee288cc4ce.png">

- **Submit** redirects you to #show, which is just a hard-coded confirmation for now:
<img width="1792" alt="Screen Shot 2020-10-29 at 12 38 49 PM" src="https://user-images.githubusercontent.com/62911934/97624327-2c376d80-19e4-11eb-8beb-f9649c8475e8.png">

- Check that the inputs were saved in Rails console and inspect `input_name` and `input_value`:

```
SurveySubmission.last.survey_submission_responses
```


### Types of changes




### Checklist:
